kroxylicious-filters/kroxylicious-record-validation/src/test/java/io/kroxylicious/proxy/filter/schema/validation/record/KeyAndValueRecordValidatorTest.java

```diff
@@ -21,38 +23,38 @@
 class KeyAndValueRecordValidatorTest {
 
     public static final String FAIL_MESSAGE = "fail";
-    public static final BytebufValidator INVALID = (buffer, length, record, isKey) -> new Result(false, FAIL_MESSAGE);
+    public static final BytebufValidator INVALID = (buffer, length, record, isKey) -> CompletableFuture.completedFuture(new Result(false, FAIL_MESSAGE));
     public static final BytebufValidator VALID = (buffer, length, record, isKey) -> Result.VALID;
 
     @Test
     void testInvalidKey() {
         RecordValidator recordValidator = keyAndValueValidator(INVALID, VALID);
-        Result validate = recordValidator.validate(Mockito.mock(Record.class));
+        Result validate = recordValidator.validate(Mockito.mock(Record.class)).toCompletableFuture().join();
         assertFalse(validate.valid());
         assertEquals("Key was invalid: " + FAIL_MESSAGE, validate.errorMessage());
```

Comment:
I think this reads better
```suggestion
      RecordValidator recordValidator = keyAndValueValidator(INVALID, VALID);
        Assertions.assertThat(recordValidator.validate(mock(Record.class)))
                .isCompletedWithValue(new Result(false, "Key was invalid: " + FAIL_MESSAGE));
```